config/config.go

```diff
@@ -743,23 +742,28 @@ var iamRoleCache = NewIAMRoleOptionsCache()
 
 // setIAMRole - extract IAM role details from Terragrunt flags block
 func setIAMRole(configString string, terragruntOptions *options.TerragruntOptions, includeFromChild *IncludeConfig, filename string) error {
-	// as key is considered HCL code and include configuration
-	var key = fmt.Sprintf("%v-%v", configString, includeFromChild)
-	var config, found = iamRoleCache.Get(key)
-	if !found {
-		iamConfig, err := TerragruntConfigFromPartialConfigString(configString, terragruntOptions, includeFromChild, filename, []PartialDecodeSectionType{TerragruntFlags})
-		if err != nil {
-			return err
+	// Prefer the IAM Role CLI args if they were passed otherwise lazily evaluate the IamRoleOptions using the config.
+	if terragruntOptions.OriginalIAMRoleOptions.RoleARN != "" {
+		terragruntOptions.IAMRoleOptions = terragruntOptions.OriginalIAMRoleOptions
+	} else {
+		// as key is considered HCL code and include configuration
+		var key = fmt.Sprintf("%v-%v", configString, includeFromChild)
+		var config, found = iamRoleCache.Get(key)
+		if !found {
+			iamConfig, err := TerragruntConfigFromPartialConfigString(configString, terragruntOptions, includeFromChild, filename, []PartialDecodeSectionType{TerragruntFlags})
+			if err != nil {
+				return err
+			}
+			config = iamConfig.GetIAMRoleOptions()
+			iamRoleCache.Put(key, config)
 		}
-		config = iamConfig.GetIAMRoleOptions()
-		iamRoleCache.Put(key, config)
-	}
-	// We merge the OriginalIAMRoleOptions into the one from the config, because the CLI passed IAMRoleOptions has
-	// precedence.
-	terragruntOptions.IAMRoleOptions = options.MergeIAMRoleOptions(
-		config,
-		terragruntOptions.OriginalIAMRoleOptions,
-	)
+		// We merge the OriginalIAMRoleOptions into the one from the config, because the CLI passed IAMRoleOptions has
+		// precedence.
+		terragruntOptions.IAMRoleOptions = options.MergeIAMRoleOptions(
+			config,
+			terragruntOptions.OriginalIAMRoleOptions,
+		)
```

Comment:
I might need to remove this, as OriginalIAMRoleOptions will likely be empty if we fall into this conditional block.

Reply:
AFAIK, OriginalIAMRoleOptions can be populated from CLI ARGs

Reply:
So I guess the other fields will be overridden unless `RoleARN` is set from the CLI args.